### DISCUSSION TITLE
Package 'testthat' was built under R version 4.0.5

### DISCUSSION THREAD

**Discussion by maxl-amberg** [ThinkR-open/golem#661] [Q&A]
Hi,
I've got a third question regarding my app called golemapp20210422 which runs perfectly fine when I use "run_dev.R" stored in the "dev" folder, but which always renders the following ERROR when I run devtools::check() in "03_deploy.R":

```
* checking tests ... ERROR
  Running 'spelling.R'
  Comparing 'spelling.Rout' to 'spelling.Rout.save' ...  Running 'testthat.R' [29s]
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  Warning message:
  package 'testthat' was built under R version 4.0.5
  > library(golemapp20210422)
  >
  > test_check("golemapp20210422")
  == Failed tests ================================================================
  -- Failure (test-golem-recommended.R:29:5): app launches -----------------------
  x$is_alive() is not TRUE
  `actual`:   FALSE
  `expected`: TRUE
  [FAIL 1 | WARN 0 | SKIP 0 | PASS 2 ]
  Error: Test failures
  Execution halted
```
@ColinFay : I am not sure whether this error message sufficies to propose a solution.
As suggested in the youtube-video "Demonstration of the {golem} package", I simply increased the time period from Sys.sleep(5) to specified Sys.sleep(30) but this does not solve the problem. Do you have an idea what the actual problem could be here?

"testthat.R" contains:
```
library(testthat)
library(golemapp20210422)
test_check("golemapp20210422")
```
"test-golem-recommended.R" contains:
```
context("golem tests")

library(golem)

test_that("app ui", {
  ui <- app_ui()
  expect_shinytaglist(ui)
})

test_that("app server", {
  server <- app_server
  expect_is(server, "function")
})

# Configure this test to fit your need
test_that(
  "app launches",{
    skip_on_cran()
    skip_on_travis()
    skip_on_appveyor()
    x <- processx::process$new(
      "R", 
      c(
        "-e", 
        "setwd('../../'); pkgload::load_all();run_app()"
      )
    )
    Sys.sleep(30)
    expect_true(x$is_alive())
    x$kill()
  }
)
```

Would it also be possible to drop the entire "test" folder, containing both "spelling.R" and "testthat.R", so that the error does not show up? I know that this would not solve the problem behind it but given that I do not want to send the app/package to CRAN or to another package manager, I thought that it would suffice that the app runs exactly as it should run when I use "run_dev.R" instead of devtools::check() in "03_deploy.R".

Thanks a lot in advance!

**Comment by ColinFay** [comment]
Hey, 

For the testthat issue, it's due to the fact that the version on CRAN was built with a developer that has a more recent version of R than you (most of the time, you can ignore that). 

For the second one, can you update `{golem}` to the latest version, delete the `test-golem-recommended` and re-run `golem::add_recommmended_tests()`? That should solve the problem. 

Cheers, 
Colin

**Reply by maxl-amberg** [reply]
Hey,

Thanks for answering both!

I have just updated `{golem}` to the latest version, deleted the `test-golem-recommended` and re-ran `golem::add_recommmended_tests()` which created a new `test-golem-recommended` containing the following:

```
test_that("app ui", {
  ui <- app_ui()
  golem::expect_shinytaglist(ui)
  # Check that formals have not been removed
  fmls <- formals(app_ui)
  for (i in c("request")){
    expect_true(i %in% names(fmls))
  }
})

test_that("app server", {
  server <- app_server
  expect_is(server, "function")
  # Check that formals have not been removed
  fmls <- formals(app_server)
  for (i in c("input", "output", "session")){
    expect_true(i %in% names(fmls))
  }
})

# Configure this test to fit your need
test_that(
  "app launches",{
    golem::expect_running(sleep = 5)
  }
)
```
However, when running `devtools::check()`, I then get a different error:

```
* checking tests ... ERROR
  Running 'spelling.R'
  Comparing 'spelling.Rout' to 'spelling.Rout.save' ...  Running 'testthat.R'
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  > 
  > test_check("golemapp20210422")
  == Skipped tests ===============================================================
  * interactive() is not TRUE (1)
  
  == Failed tests ================================================================
  -- Failure (test-golem-recommended.R:7:5): app ui ------------------------------
  i %in% names(fmls) is not TRUE
  
  `actual`:   FALSE
  `expected`: TRUE 
  
  [ FAIL 1 | WARN 0 | SKIP 1 | PASS 5 ]
  Error: Test failures
  Execution halted
```

Best,
Maxl
